storage/sealer/fsutil/statfs_unix.go

```diff
@@ -15,12 +20,113 @@ func Statfs(path string) (FsStat, error) {
 		return FsStat{}, xerrors.Errorf("statfs: %w", err)
 	}
 
+	used := int64(stat.Blocks-stat.Bavail) * int64(stat.Bsize)
+	available := int64(stat.Bavail) * int64(stat.Bsize)
+
+	// Filesystems like ZFS can contain sub-volumes which are distinct filesystems, but use the same pool of space.
+	// To accurately account for space used in sub-volumes, accumulate the used space in each filesystem mounted under the path.
+	mounts, err := getMountsUnder(path)
+	if err != nil {
+		log.Warnf("could not get all mounts under %s: %e", path, err)
+	}
+
+	for _, mount := range mounts {
+		var mountStat syscall.Statfs_t
+		if err := syscall.Statfs(mount, &mountStat); err != nil {
+			log.Warnf("could not get stats for mount %s: %e", mount, err)
+			continue
+		}
+		used += int64(mountStat.Blocks-mountStat.Bavail) * int64(mountStat.Bsize)
```

Comment:
I feel like we'll need to make this configurable somehow, or at the very least a bit smarter.

* What if there are separate filesystems here - in that case we'd want to add those to `available` as well
* Does this work with btrfs subvolumes the same way? (I'm forgetting how those work now, but would be nice if they also worked with this)

Reply:
You're right that this would not be accurate if the filesystems do not share available space. I'm not sure how that can be accurately accounted for. I don't know of any way to reliably determine if two mounts share a pool of available space. You could guess that they do if the available space is the same. But with reservations and quotas you could have a situation where the available space is different, but still shared.

My thinking was this wouldn't be any worse than it works today, and would improve things for one specific use case, even if it doesn't fix every possible situation.

I don't have any experience with btrfs, so I can't comment on that.